### PR TITLE
sink(ticdc): Kafka client hide sarama config

### LINKED ISSUE #8076: Kafka client hide sarama config
Kafka client hide sarama config

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #8076

### What is changed and how it works?
Kafka client hide sarama config

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/sink/mq/producer/kafka/kafka.go [hunk 1]
```diff
@@ -369,7 +368,6 @@ func NewKafkaSaramaProducer(
 func AdjustConfig(
```
Comment (3AceShowHand) [inline cdc/sink/mq/producer/kafka/kafka.go:368]: It looks can be renamed to `AdjustOptions`.

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/sink/mq/producer/kafka/kafka.go (hunk 1 above)
Comment (sdojjy) [reply cdc/sink/mq/producer/kafka/kafka.go]: done

**Review by asddongmen — APPROVED** [review]

**Pull request merged**